pandas/core/internals.py

```diff
@@ -3670,19 +3670,25 @@ def copy(self, deep=True, mgr=None):
         return self.apply('copy', axes=new_axes, deep=deep,
                           do_integrity_check=False)
 
-    def as_matrix(self, items=None):
+    def as_matrix(self, axis=1, items=None):
         if len(self.blocks) == 0:
             return np.empty(self.shape, dtype=float)
 
+        other_axis = abs(axis-1)
         if items is not None:
-            mgr = self.reindex_axis(items, axis=0)
+            mgr = self.reindex_axis(items, axis=other_axis)
         else:
             mgr = self
 
         if self._is_single_block or not self.is_mixed_type:
-            return mgr.blocks[0].get_values()
+            arr = mgr.blocks[0].get_values()
         else:
```

Comment:
change this to ``transpose`` instead. should be straightforward from here.

Reply:
Thanks a lot, ``transpose`` is of course much better than axis.

The issue was actually in the ``if len(self.blocks) == 0:`` block, as the empty array also must be transposed. 

Everything is green now locally and I've pushed that upstream.